kolibri/deployment/default/custom_django_cache.py

```diff
@@ -13,6 +13,32 @@ class CustomDjangoCache(DjangoCache):
     https://github.com/grantjenks/python-diskcache/blob/v4.1.0/diskcache/djangocache.py
     """
 
+    ERRORS_TO_HANDLE = (sqlite3.OperationalError, AssertionError)
+
+    def try_execute_return_none_on_error(self, method, *args, **kwargs):
```

Comment:
I think you could simplify this all somewhat by having a single method like this:
```
def try_execute(self, method_name, error_return_value, *args, **kwargs):
    try:
        method = getattr(super(CustomDjangoCache, self), method_name)
        if method is None:
            raise ValueError("{method_name} is not a valid method".format(method_name=method_name))
        return method(*args, **kwargs)
    except self.ERRORS_TO_HANDLE:
        return error_return_value
```

That way you can guarantee you're using the Py2.7 compatible form of super, and you don't need to define 4 different functions that are doing the same thing.

If a function doesn't return a value explicitly in Python, it returns `None`, so you can pass `None` as the `error_return_value` in those cases.

Reply:
Thanks for the code snippet, indeed this thought crossed my mind, but i was confused about `error_return_value` having types. Now it makes more sense to me